### PR TITLE
Feature/8193 create product txt file & add exchangeset part2

### PR DESCRIPTION
**Pull request by pratibh14837**
Code Changes for package update for product.txt file.
Also we have done changes for adding delay while checking status of batch in same PR.
Commit of batch takes time and as a result it was returning message "Exchange set is not created" even though exchange set is uploaded to FSS successfully.
We didn't added this code earlier as there was enterprise event grid issue.
As that issue is fixed at start of this sprint, we have added that delay code now.

### CODE REVIEW

**Review by janvi13603 — APPROVED** [review]
Changes looks good.

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs [hunk 1]
```diff
@@ -353,24 +355,43 @@ public async Task<bool> CreateExchangeSetFile(string batchId, string correlation
...
+                 await Task.Delay(fileShareServiceConfig.Value.BatchCommitDelayTimeInMilliseconds);
+                }
+                if (batchStatus != BatchStatus.Committed)
+                {
+                    watch.Stop();
+                    logger.LogError(EventIds.BatchCommitTimeout.ToEventId(), "Batch Commit Status timedout with BatchStatus:{batchStatus} for BatchId:{batchId} and _X-Correlation-ID:{CorrelationId}", batchStatus, batchStatusMetaData.BatchId, correlationId);
+                    return batchStatus;
+                }
+                watch.Stop();
             }
```
Comment (martyn-fewtrell) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs:395]: If isUploadCommitBatchCompleted is False will we return BatchStatus.CommitInProgress?

Is this correct?

Should we set BatchStatus to something more appropriate (Failed?) before returning the value?

**Review by martyn-fewtrell — COMMENTED** [review]
Just one comment to consider.

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs (hunk 1 above)
Comment (pratibh14837) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs:395]: Resolved.

**Review by martyn-fewtrell — APPROVED** [review]
Thanks for the update

**Pull request merged**